infra/terraform/modules/data_access/users/users.py

```diff
@@ -0,0 +1,75 @@
+import json
+import os
+
+import boto3
+
+
+'''
+Environment variables:
+ - STAGE, e.g. "dev", "alpha", etc...
+ - SAML_PROVIDER_ARN, SAML provider used to login
+'''
+
+
+def create_user_role(event, context):
+    """
+    Creates the role for the given user
+
+    event = {"username": "alice"}
+    """
+    trust_relationship = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": os.environ["SAML_PROVIDER_ARN"],
+                },
+                "Action": "sts:AssumeRoleWithSAML",
+                "Condition": {
+                    "StringEquals": {
+                        "SAML:aud": "https://signin.aws.amazon.com/saml"
+                    }
+                }
+            }
+        ]
+    }
+
+    client = boto3.client("iam")
+    client.create_role(
+        RoleName=role_name(event["username"]),
+        Path="/users/",
+        AssumeRolePolicyDocument=json.dumps(trust_relationship),
```

Comment:
I missed this, I'll change.

On a related note, do you think we need to pass `"utf8"` all the time considering it's the default encoding and it's a good default? It feels like we clutter the code without much benefit (and less code is usually a good thing).